app/electron/main.ts

```diff
@@ -1,33 +1,33 @@
-const { app, BrowserWindow, Menu } = require('electron');
-const { spawn } = require('child_process');
-const path = require('path');
-const url = require('url');
-const { autoUpdater } = require('electron-updater');
-const log = require('electron-log');
-const open = require('open');
-
-const yargs = require('yargs');
+import { app, BrowserWindow, Menu, MenuItem } from 'electron';
+import { spawn, ChildProcessWithoutNullStreams } from 'child_process';
+import path from 'path';
+import url from 'url';
+import { autoUpdater } from 'electron-updater';
+import  log from 'electron-log';
+import open from 'open';
+import yargs from 'yargs';
+import { MenuItemConstructorOptions } from 'electron/main';
+
 const args = yargs.option('headless', {
   describe: 'Open Headlamp in the default web browser instead of its app window'
 }).parse();
 const isHeadlessMode = args.headless;
 const defaultPort = 4466;
 
-function startServer(flags = []) {
+function startServer(flags:string[] = []):ChildProcessWithoutNullStreams {
   const serverFilePath = path.join(process.resourcesPath, './electron/server');
   return spawn(serverFilePath, [...flags], {shell: true});
 }
 
-let serverProcess = null;
-let intentionalQuit = false;
-let serverProcessQuit = false;
+let serverProcess: ChildProcessWithoutNullStreams | null;
+let intentionalQuit:boolean;
+let serverProcessQuit:boolean;
 
 function startElecron() {
-  autoUpdater.logger = log;
-  autoUpdater.logger.transports.file.level = 'info';
+  log.transports.file.level = 'info';
   log.info('App starting...');
 
-  let mainWindow;
+  let mainWindow: BrowserWindow | null;
```

Comment:
I think this should be `BrowserWindow | undefined` rather than null?

Reply:
I think for objects, having null type makes more sense